### PR TITLE
fix warning highlight resizing

### PR DESCRIPTION
**Pull request by riknoll**
fixes https://github.com/microsoft/pxt-microbit/issues/5877

when setting the warning message on a block it changes the size of the block (the icon adds width).

this pr makes sure to resize the highlight border post render once the icon is visible.

### CODE REVIEW

File: webapp/src/blocks.tsx [hunk 1]
```diff
@@ -1956,9 +1956,13 @@ function clearTemporaryAssetBlockData(workspace: Blockly.Workspace) {
     forEachImageField(workspace, field => field.clearTemporaryAssetData());
 }
 
-function setHighlightWarning(block: Blockly.BlockSvg, enabled: boolean) {
+async function setHighlightWarning(block: Blockly.BlockSvg, enabled: boolean) {
```
Comment (srietkerk) [inline webapp/src/blocks.tsx:1959]:
```suggestion
async function setHighlightWarningAsync(block: Blockly.BlockSvg, enabled: boolean) {
```

File: webapp/src/blocks.tsx [hunk 2]
```diff
@@ -1956,9 +1956,13 @@ function clearTemporaryAssetBlockData(workspace: Blockly.Workspace) {
...
     (block.pathObject as PathObject).setHasError(enabled);
     block.setHighlighted(enabled);
```
Comment (srietkerk) [inline webapp/src/blocks.tsx:1961]: Does setting the highlight and then resizing make it so there's a snapping effect for the highlight? Or is it fast enough that there's no visual difference?

Comment (riknoll) [reply webapp/src/blocks.tsx:1961]: there's no visual difference

File: webapp/src/blocks.tsx (hunk 1 above)
Comment (kimprice) [inline webapp/src/blocks.tsx:1959]: Should we follow convention here and name it `setHighlightWarningAsync`?

Comment (riknoll) [reply webapp/src/blocks.tsx:1959]: i renamed it

**Review by jwunderl — APPROVED** [review]

**Pull request merged**